java/com/google/flatbuffers/TableVector.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2019 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.flatbuffers;
+
+import java.nio.ByteBuffer;
+
+/// @cond FLATBUFFERS_INTERNAL
+
+/**
+ * All vector of tables in the generated code derive from this class, and add their own accessors.
+ */
+public class TableVector {
```

Comment:
There doesn't appear anything specific to tables in this class, it could easily work for a vector of doubles or whatever too. Why not add support for vectors of other types while we're at it? I'd think that the gain of only looking up the vector just once is most pronounced with iterating a scalar vector.

Reply:
There are a number of conceptual issues. The current API for scalar and string vectors is different from the API for structure and table vectors in access objects:
1) All of these types are built-in and we need to manually create a vector object for each type of scalar (integer, fractional, byte, etc.) in com.google.flatbuffers, but not in the generated code.
2) The generator must add mutability methods to access objects for scalar type vectors if the corresponding flag is set. This conflicts with clause 1. The vectors of tables and structures are immutable.

Reply:
Or we will have to generate code for the com.google.flatbuffers package along with the code from the schema. This will be similar to the work of the C ++ compiler with templates. The file structure of the generated code will look something like this:
`foo/bar/` - classes from schema
`com/google/flatbuffers/` - classes for vectors of built-in types

Reply:
@aardappel The method with the generated classes will be conflicted when several schemes are used in the same project with different values of the `--gen-mutable` flag.

Reply:
How is this problem solved in C ++ code?

Reply:
But that's what I am saying: the current `TableVector` could actually work for other types, and would not need a copy per type.

It be fine to leave out the mutation functions for this particular class if it makes things complicated, since if someone  wants to mutate, they can still use the existing functions.

Generally it be nicer to not add copies of such classes to generated code whenever possible (efficiency wise).

In C++, the `Vector` template always has mutation function, irrespective of whether any mutation generated code exists.